### PR TITLE
Added events for cluster autoscaler.

### PR DESCRIPTION
**Pull request by jszczepkowski**
Added events for cluster autoscaler.

### CODE REVIEW

File: cluster-autoscaler/estimator/estimator.go [hunk 1]
```diff
@@ -29,16 +29,18 @@ import (
 // It will never overestimate the number of nodes but is quite likekly to provide a number that
 // is too small.
 type BasicNodeEstimator struct {
-	count     int
-	cpuSum    resource.Quantity
-	memorySum resource.Quantity
-	portSum   map[int32]int
+	count          int
```
Comment (piosz) [inline cluster-autoscaler/estimator/estimator.go]: once you introduced PodsToSchedule we can get rid off of count.

File: cluster-autoscaler/scale_up.go [hunk 2]
```diff
@@ -125,7 +128,17 @@ func ScaleUp(unschedulablePods []*kube_api.Pod, nodes []*kube_api.Node, migConfi
 		if err := gceManager.SetMigSize(bestOption.migConfig, newSize); err != nil {
 			return false, fmt.Errorf("failed to set MIG size: %v", err)
 		}
+
+		for pod := range bestOption.estimator.PodsToSchedule {
+			recorder.Eventf(pod, kube_api.EventTypeNormal, "CausedScaleUp",
+				"pod caused scale-up, mig: %s, sizes (current/new): %d/%d", bestOption.migConfig.Name, currentSize, newSize)
```
Comment (piosz) [inline cluster-autoscaler/scale_up.go]: /s/caused/triggered?

File: cluster-autoscaler/scale_up.go [hunk 3]
```diff
@@ -125,7 +128,17 @@ func ScaleUp(unschedulablePods []*kube_api.Pod, nodes []*kube_api.Node, migConfi
...
+		}
+
 		return true, nil
 	}
+	for pod := range podsRemainUnshedulable {
+		recorder.Event(pod, kube_api.EventTypeNormal, "DoseNotCauseScaleUp", "pod doesn't cause scale-up (doesn't match scale-up predicate)")
```
Comment (piosz) [inline cluster-autoscaler/scale_up.go]: /s/Dose/Does

Comment (piosz) [inline cluster-autoscaler/scale_up.go]: The message is weird for me. IMO we should say that we tried to help this pod be we wasn't able to do it.

File: cluster-autoscaler/estimator/estimator.go [hunk 4]
```diff
@@ -29,16 +29,18 @@ import (
...
+	cpuSum         resource.Quantity
+	memorySum      resource.Quantity
+	portSum        map[int32]int
+	PodsToSchedule map[*kube_api.Pod]struct{}
```
Comment (piosz) [inline cluster-autoscaler/estimator/estimator.go]: optional: better naming here

Comment (jszczepkowski) [inline cluster-autoscaler/estimator/estimator.go]: any suggestion?

File: cluster-autoscaler/scale_up.go (hunk 2 above)
Comment (jszczepkowski) [inline cluster-autoscaler/scale_up.go]: done

File: cluster-autoscaler/estimator/estimator.go (hunk 1 above)
Comment (jszczepkowski) [inline cluster-autoscaler/estimator/estimator.go]: done

File: cluster-autoscaler/scale_up.go (hunk 3 above)
Comment (jszczepkowski) [inline cluster-autoscaler/scale_up.go]: done

**Comment by jszczepkowski** [comment]
PTAL

**Comment by piosz** [comment]
LGTM

**Comment by piosz** [comment]
cc @mwielgus @fgrzadkowski

**Comment by piosz** [comment]
```
# k8s.io/contrib/cluster-autoscaler/estimator
estimator/estimator_test.go:55: estimator.count undefined (type *BasicNodeEstimator has no field or method count)
```
please fix

**Pull request merged**